### ISSUE TITLE
[AIR] Checkpoint model and report metric at different steps

### ISSUE THREAD

**Issue by bveeramani** [ray-project/ray#31213]
### Description

Update `ray.air.integrations.keras.Callback` to let users checkpoint and report metrics at different frequencies.

We could do this by splitting `Callback` into two distinct callbacks:

```
CheckpointModel(on="epoch_end)
ReportMetrics(on=("on_train_batch_end", "on_test_batch_end"), frequency=4)
```

### Use case

I want to checkpoint every epoch and report metrics to W&B every few batches. I can sort-of achieve this by using the `frequency` parameter, but this isn't ideal because:
1. I have to compute the correct `frequency`
2. I have the report metrics every batch (as opposed to every few)

```
Callback(on=("on_train_batch_end", "on_test_batch_end"), frequency=???)
```
